pkg/sql/schemachanger/scdeps/sctestdeps/test_deps.go

```diff
@@ -560,9 +560,18 @@ func (s *TestState) MustReadDescriptor(ctx context.Context, id descpb.ID) catalo
 	return desc
 }
 
+// AddSyntheticDescriptor is part of the scexec.Catalog interface.
+func (s *TestState) AddSyntheticDescriptor(desc catalog.MutableDescriptor) {
+	s.LogSideEffectf("added synthetic descriptor: %v", desc)
```

Comment:
Can we instead print the yaml diff between the synthetic descriptor and the existing immutable descriptor if it exists?